scipy/interpolate/_interpolate.py

```diff
@@ -17,6 +17,9 @@
 from .interpnd import _ndim_coords_from_arrays
 from ._bsplines import make_interp_spline, BSpline
 
+# deprecated import to be removed in SciPy 1.13.0
+import itertools  # noqa
```

Comment:
I don't understand this comment. `itertools` is a Python stdlib module, not one from scipy?

Reply:
at some point it was exposed and added to `__all__` (probably unintentionally) so just followed the same pattern as in the other prs. Can remove straight away from here and `__all__` if you'd prefer

Reply:
OK that's fine, but then the comment IMO should have a bit more meat, e.g.
```suggestion
# even though this is a stdlib module, it got accidentally exposed in __all__ in the past.
# it os now deprecated and scheduled to be removed in SciPy 1.13.0
import itertools  # noqa
```